core/chains/evm/evm_txm.go

```diff
@@ -0,0 +1,54 @@
+package evm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/smartcontractkit/sqlx"
+
+	evmclient "github.com/smartcontractkit/chainlink/core/chains/evm/client"
+	evmconfig "github.com/smartcontractkit/chainlink/core/chains/evm/config"
+	httypes "github.com/smartcontractkit/chainlink/core/chains/evm/headtracker/types"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/core/chains/evm/txmgr"
+	evmtypes "github.com/smartcontractkit/chainlink/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/core/logger"
+)
+
+// GenericTxManager is type alias for txmgr.TxManager.
+// This is necessary because Golang doesn't allow embedding
+// txmgr.TxManager directly inside evmTxm struct.
+type GenericTxManager = txmgr.TxManager
+
+var _ httypes.HeadTrackable = &evmTxm{}
+
+// evmTxm is an evm wrapper over the generic TxManager interface
+type evmTxm struct {
+	httypes.HeadTrackable
+	GenericTxManager
+}
```

Comment:
I think we should be able to use the TxManager struct embedded/anonymous.
```
type evmTxm struct {
	httypes.HeadTrackable
	txmgr.TxManager
}
```

Reply:
I think the above doesn't work because you can't do this
```
func (e evmTxm) OnNewLongestChain(ctx context.Context, head *evmtypes.Head) {
	e.txmgr.TxManager.OnNewLongestChain(ctx, head)
}
```

Reply:
hmm, i think you can directly call on the `TxManager`.  unless I'm missing something? 

```
type evmTxm struct {
	httypes.HeadTrackable
	txmgr.TxManager
}

func (e evmTxm) OnNewLongestChain(ctx context.Context, head *evmtypes.Head) {
	e.TxManager.OnNewLongestChain(ctx, head)
}
```

Reply:
Oh thanks for the context, ill check it out

Reply:
Another way of fixing without the type alias, but what you have here is good too :+1: , thanks I learned something new :)
https://gotipplay.golang.org/p/csPUU1k1XC1

Reply:
The failing case is trying to use the package in the field name. Won't it work with just the type?

Reply:
@prashantkumar1982  - can we remove the comments 18-21 please? that got me confused - I deleted my comments by mistake earlier.
@jmank88 - It should work
@samsondav  - yea lines 18-21.

Reply:
@samsondav , He changed it, it looks good now!